src/main/java/cuchaz/enigma/network/packet/UserListS2CPacket.java

```diff
@@ -0,0 +1,51 @@
+package cuchaz.enigma.network.packet;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import cuchaz.enigma.gui.GuiController;
+
+public class UserListS2CPacket implements Packet<GuiController> {
+
+	private List<String> users;
+
+	UserListS2CPacket() {
+	}
+
+	public UserListS2CPacket(List<String> users) {
+		this.users = users;
+	}
+
+	@Override
+	public void read(DataInput input) throws IOException {
+		int len = input.readInt();
+		if (len == 0) {
+			users = Collections.emptyList();
+		} else if (len == 1) {
+			users = Collections.singletonList(input.readUTF());
+		} else {
+			users = new ArrayList<>(len);
+			for (int i = 0; i < len; i++) {
+				users.add(input.readUTF());
+			}
+		}
```

Comment:
I'm not sure the tiny amount this optimizes things justifies this complexity. It's okay to create small temporary objects in Java, such as ArrayLists of size 0.